Add vision support for OpenAI models in agent panel

07rjain opened the pull request:
## Summary
  Add vision support for OpenAI models in agent panel

  Enable image processing capabilities for vision-capable OpenAI models:
  - GPT-4o, GPT-4o-mini
  - GPT-4.1, GPT-4.1-mini, GPT-4.1-nano
  - O1, O3, O4-mini

  **Updated**: Refactored implementation to eliminate redundant `supports_vision()`
  wrapper method and implement vision logic directly in the `supports_images()` trait
  method, following established patterns used by other static providers.

  ## Changes
  - ✅ Direct implementation in `supports_images()` without unnecessary wrapper methods
  - ✅ Consistent with Anthropic/Google provider patterns
  - ✅ Cleaner, more maintainable code

  This brings OpenAI vision support in line with GitHub Copilot, allowing users to send
   images to vision-capable models through the agent panel.

07rjain committed:
Add vision support for OpenAI models in agent panel

Enable image processing capabilities for OpenAI models that support vision:
- GPT-4.1, GPT-4.1-mini, GPT-4.1-nano
- GPT-4o, GPT-4o-mini
- O1, O3, O4-mini

This brings OpenAI vision support in line with GitHub Copilot, allowing
users to send images to vision-capable models through the agent panel.

07rjain commented:
@cla-bot check

versecafe reviewed (commented):
is there a reason `supports_vision` exists when it matches the type sig of `supports_images` and is just wrappered directly by it?

07rjain committed:
Refactor: Remove redundant supports_vision method

Remove the unnecessary supports_vision() wrapper method and implement
vision capabilities directly in supports_images() trait method.

This eliminates code duplication and follows the established pattern
used by other static providers in the codebase (Anthropic, Google).

Addresses feedback about redundant wrapper method with identical
type signature and functionality.

07rjain commented:
@versecafe 

You're absolutely right! That was unnecessary duplication.

The `supports_vision()` method was indeed redundant since it had identical functionality to `supports_images()`. I've refactored the implementation to eliminate that wrapper and put the vision logic directly in the `supports_images()` trait method, which is cleaner and follows the same pattern used by other static providers like Anthropic and Google.

Thanks for catching that! 🙏

versecafe commented on crates/open_ai/src/open_ai.rs:
```diff
@@ -201,6 +201,7 @@ impl Model {
             Self::O1 | Self::O3 | Self::O3Mini | Self::O4Mini | Model::Custom { .. } => false,
         }
     }
+
```
uneeded white space diff here

07rjain committed:
Fix unnecessary whitespace in open_ai.rs

Remove extra empty line after impl block as requested in PR review.

07rjain commented:
Fixed the unnecessary whitespace diff in `crates/open_ai/src/open_ai.rs`. Thank you for the review!

07rjain committed:
Merge branch 'zed-industries:main' into add-openai-vision-support

07rjain commented:
Request is already merged [#36047 ](https://github.com/zed-industries/zed/pull/36047), no need to merge this one!! thanks
